library/common/config/config.cc

```diff
@@ -326,6 +326,7 @@ R"(
         dns_cache_config: *dns_cache_config
     transport_socket: *base_tls_socket
     upstream_connection_options: &upstream_opts
+      set_local_interface_name_on_upstream_connections: true
```

Comment:
How much of a perf impact does this have for EM use cases compared to turning it off? Would it be worth offering as a config option or turning off when log level is set to some values?

Reply:
https://github.com/envoyproxy/envoy-mobile/issues/2111 good point. Alternatively there is further work in upstream envoy that can make this functionality more performant. 

For this PR though I'm biased to leave it on by default as that has been the case for all experiments we've run in the last quarter.

Reply:
Sure, I'm ok with that too, but was curious if you had measured the impact on us.